arch/sim/Kconfig

```diff
@@ -15,18 +15,22 @@ config HOST_X86_64
 	select ARCH_HAVE_STACKCHECK
 	select LIBC_ARCH_ELF_64BIT if LIBC_ARCH_ELF && !SIM_M32
 	select ARCH_HAVE_MATH_H
+	select ARCH_TOOLCHAIN_GNU
```

Comment:
why need select the toolchain

Reply:
kasan/Makefile:
```
ifeq ($(CONFIG_ARCH_TOOLCHAIN_GNU),y)
  CFLAGS += -fno-sanitize=kernel-address
  ifeq ($(CONFIG_LTO_NONE),n)
    CFLAGS += -fno-lto
  endif
endif
```If SIM does not select and mm does not turn off instrumentation, it will cause recursion.

Reply:
@W-M-R but it doesn't sound correct to select a fixed toolchain, maybe the recursion issue should be fixed other way. What if the developer decide to use a toolchain different from GNU/GCC ?

Reply:
sim doesn't support user switch to a different toolchain. it can be only compiled with the host default one. All of them compatible with gcc.

Reply:
Thank you Xiang. But yet the solution seems like a "workaround", because if someone decide to add support to build sim using other toolchain this "fix" needs to be fixed again. So, maybe it is better to fix the root cause :-)

Reply:
In that case, the contributor just need move the selection to the toolchain which compatible with gcc. It's hard to find a better place to select ARCH_TOOLCHAIN_GNU before the toolchain selection is added to sim.